pymdownx/magiclink.py

```diff
@@ -48,7 +48,7 @@
 RE_LINK = r'''(?xi)
 (?P<link>
     (?:(?<=\b)|(?<=_))(?:
-        (?:ht|f)tps?://(?:(?:[^_\W][-\w]*(?:\.[-\w.]+)+)|localhost)|  # (http|ftp)://
+        (?:ht|f)tps?://(?:[^_\W][-\w]*(?:[^ \d][-\w.]+)+)|  # (http|ftp)://
```

Comment:
This should probably apply to the `www.hostname` as well.

With that said, can you explain the approach of changing `\.` to `[^ \d]`?

Reply:
Specifically why that pattern? Why are we avoiding `\d`?

Reply:
Actually wondering if it would be better just to make the dot portion optional...

Reply:
Initially I went to `\.?` but that resulted in the negative test for `http://3628126748` failing (i.e. that became a link) so I went to `\D` but that resulted in `http://foo.bar/foo(bar)baz quux` being matched rather than excluding the `quux` being excluded. So I settled on `[^ \d]`.

I'm not 100% comfortable with that blacklist based on existing tests as an approach, but I'm not sure how else to prevent `http://3628126748` matching.

`www.hostname` does match with this regex, is that what we want?

Reply:
I unquoted the numeric link in my message, and it looks like GitHub allows that string. Looks like that test has been around since https://github.com/facelessuser/pymdown-extensions/commit/15e481e8bf757e8d8199db2c398e4bf2f9ce1439 I wonder if we should just let it be a URL?

Reply:
No, I didn't mean `\.?`, I meant the entire dot portion of the pattern.  I went to code up the example, but then I needed to test it...so here is a pull: https://github.com/facelessuser/pymdown-extensions/pull/796.

As far as the failing test is concerned, I guess we could relax it a bit and allow numerical host name like GitHub and friends does.